test/envoye2e/driver/envoy.go

```diff
@@ -84,6 +90,11 @@ func (e *Envoy) Run(p *Params) error {
 	envoyPath := filepath.Join(env.GetDefaultEnvoyBin(), "envoy")
 	if path, exists := os.LookupEnv("ENVOY_PATH"); exists {
 		envoyPath = path
+	} else if _, err := os.Stat(envoyPath); os.IsNotExist(err) && e.Version != "" {
+		envoyPath, err = downloadEnvoy(e.Version)
```

Comment:
When e.version is specified, we want to make sure that it is the correct envoy binary.

Reply:
I make the field name more clear that it is about version of envoy to download. I want to avoid the complexity of envoy version check here.